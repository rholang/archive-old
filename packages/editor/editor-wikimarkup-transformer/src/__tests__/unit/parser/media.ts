import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - Images and attachments', () => {
  const testCases: Array<[string, string]> = [
    ['should find images in the text', '!image.png!'],
    [
      'should find absolute URL images and convert it to external media item',
      '!http://www.host.com/image.gif!',
    ],
    [
      'should find attachments with attributes',
      '!quicktime.mov|width=300,height=400!',
    ],
    [
      'should find attachments in multiline string',
      `this is a line of text
!image.gif|align=right, vspace=4!
yep`,
    ],
    [
      'should find images in a multiline string with return symbols',
      '!Kapture 2018-04-04 at 16.36.13.gif!\r\n\r\nFoo',
    ],
    [
      '[CS-216] should parse media filename with "(" and ")"',
      '!Screen Shot (9db1eca8-8257-4763-92fb-e6417f9e34c9).jpeg|thumbnail!',
    ],
    [
      '[CS-1404] should parse attachments as one media group',
      '[^a-doc (jadsjdasjadsjkdasjk).pdf][^not-empty (askjsajnkjknads).txt]',
    ],
    [
      '[CS-1404] should parse attachments separated by a single new line as one media group',
      '[^a-doc (jadsjdasjadsjkdasjk).pdf]\r\n[^not-empty (askjsajnkjknads).txt]',
    ],
    [
      '[CS-1404] should parse attachments separated by any number of spaces as one media group',
      '[^a-doc (jadsjdasjadsjkdasjk).pdf]   [^not-empty (askjsajnkjknads).txt]',
    ],
    [
      '[CS-1404] should parse attachments separated by non-space character as separate media groups',
      '[^a-doc (jadsjdasjadsjkdasjk).pdf]a[^not-empty (askjsajnkjknads).txt]',
    ],
    [
      '[CS-1404] should parse attachments separated by multiple new lines as separate media groups',
      '[^a-doc (jadsjdasjadsjkdasjk).pdf]\r\n\r\n[^not-empty (askjsajnkjknads).txt]',
    ],
    [
      '[CS-1404] should parse list of multiple attachments separated by multiple new lines as separate media groups with multiple child media elements',
      '[^a-doc (jadsjdasjadsjkdasjk).pdf]\r\n[^not-empty (askjsajnkjknads).txt]\r\n\r\n[^a-doc (jadsjdasjadsjkdasjk).pdf]\r\n[^not-empty (askjsajnkjknads).txt]',
    ],
    [
      '[CS-1404] should parse attachments inside tables as single media group',
      '|colum 1 [^a-doc (jadsjdasjadsjkdasjk).pdf]\r\n[^not-empty (askjsajnkjknads).txt]|column 2|',
    ],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    it(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
