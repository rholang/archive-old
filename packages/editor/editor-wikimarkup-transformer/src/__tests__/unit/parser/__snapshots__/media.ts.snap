// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - Images and attachments [CS-216] should parse media filename with "(" and ")" 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": 183,
            "id": "Screen Shot (9db1eca8-8257-4763-92fb-e6417f9e34c9).jpeg",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 200,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments [CS-1404] should parse attachments as one media group 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "a-doc (jadsjdasjadsjkdasjk).pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "not-empty (askjsajnkjknads).txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments [CS-1404] should parse attachments inside tables as single media group 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "colum 1 ",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
                Object {
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "__contextId": null,
                        "__displayType": null,
                        "__fileMimeType": null,
                        "__fileName": null,
                        "__fileSize": null,
                        "collection": "",
                        "height": null,
                        "id": "a-doc (jadsjdasjadsjkdasjk).pdf",
                        "occurrenceKey": null,
                        "type": "file",
                        "url": null,
                        "width": null,
                      },
                      "type": "media",
                    },
                    Object {
                      "attrs": Object {
                        "__contextId": null,
                        "__displayType": null,
                        "__fileMimeType": null,
                        "__fileName": null,
                        "__fileSize": null,
                        "collection": "",
                        "height": null,
                        "id": "not-empty (askjsajnkjknads).txt",
                        "occurrenceKey": null,
                        "type": "file",
                        "url": null,
                        "width": null,
                      },
                      "type": "media",
                    },
                  ],
                  "type": "mediaGroup",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "column 2",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments [CS-1404] should parse attachments separated by a single new line as one media group 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "a-doc (jadsjdasjadsjkdasjk).pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "not-empty (askjsajnkjknads).txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments [CS-1404] should parse attachments separated by any number of spaces as one media group 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "a-doc (jadsjdasjadsjkdasjk).pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "not-empty (askjsajnkjknads).txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments [CS-1404] should parse attachments separated by multiple new lines as separate media groups 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "a-doc (jadsjdasjadsjkdasjk).pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "not-empty (askjsajnkjknads).txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments [CS-1404] should parse attachments separated by non-space character as separate media groups 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "a-doc (jadsjdasjadsjkdasjk).pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
    Object {
      "content": Array [
        Object {
          "text": "a",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "not-empty (askjsajnkjknads).txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments [CS-1404] should parse list of multiple attachments separated by multiple new lines as separate media groups with multiple child media elements 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "a-doc (jadsjdasjadsjkdasjk).pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "not-empty (askjsajnkjknads).txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "a-doc (jadsjdasjadsjkdasjk).pdf",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": null,
            "id": "not-empty (askjsajnkjknads).txt",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments should find absolute URL images and convert it to external media item 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": null,
            "height": 183,
            "id": "",
            "occurrenceKey": null,
            "type": "external",
            "url": "http://www.host.com/image.gif",
            "width": 200,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments should find attachments in multiline string 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "this is a line of text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": 183,
            "id": "image.gif",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 200,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "content": Array [
        Object {
          "text": "yep",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments should find attachments with attributes 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": 400,
            "id": "quicktime.mov",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 300,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments should find images in a multiline string with return symbols 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": 183,
            "id": "Kapture 2018-04-04 at 16.36.13.gif",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 200,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "content": Array [
        Object {
          "text": "Foo",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Images and attachments should find images in the text 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": null,
            "__displayType": null,
            "__fileMimeType": null,
            "__fileName": null,
            "__fileSize": null,
            "collection": "",
            "height": 183,
            "id": "image.png",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 200,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
  ],
  "type": "doc",
}
`;
